apps/studio/src/features/dashboard/components/IndexpageRow/IndexpageRow.tsx

```diff
@@ -0,0 +1,94 @@
+import Link from "next/link"
+import { HStack, IconButton, Skeleton, Text, VStack } from "@chakra-ui/react"
+import { Badge, BadgeLeftIcon } from "@opengovsg/design-system-react"
+import { ResourceState } from "~prisma/generated/generatedEnums"
+import { BiChevronRight, BiFile, BiSolidCircle } from "react-icons/bi"
+
+import { trpc } from "~/utils/trpc"
+
+interface IndexpageRowProps {
+  siteId: number
+  resourceId: string
+}
+
+export const IndexpageRow = ({ siteId, resourceId }: IndexpageRowProps) => {
```

Comment:
question: thoughts on using `z.infer<typeof getIndexpageSchema>` as the type instead of redefining it as `IndexpageRowProps`? since the values passed in will be used to pass into `folder.getIndexpage`

Reply:
good call, updated